crates/biome_js_semantic/src/events.rs

```diff
@@ -280,6 +281,13 @@ struct Scope {
     /// If this scope allows declarations to be hoisted to parent scope or not.
     hoisting: ScopeHoisting,
     is_in_strict_mode: bool,
+    is_ambient: bool,
+}
+
+#[derive(Debug, Default)]
+struct ScopeOptions {
+    is_closure: bool,
+    implies_strict_mode: bool,
```

Comment:
Can we document all the new types an fields?

Reply:
They are not new, I just encapsulated them in a type to make it simpler to read.

Reply:
I added some doctests.